### PR TITLE
refactor some code in modifyResponseMonths and ensure that the last mo…

### PR DESCRIPTION
**Pull request by HridoyRoy**
…nth comparison with end is comparing end of month with end of month

### CODE REVIEW

File: vault/activity_log.go [hunk 1]
```diff
@@ -1737,13 +1737,11 @@ func modifyResponseMonths(months []*ResponseMonth, start time.Time, end time.Tim
 	}
 	start = timeutil.StartOfMonth(start)
 	if timeutil.IsCurrentMonth(end, time.Now().UTC()) {
-		end = timeutil.StartOfMonth(end).AddDate(0, -1, 0)
+		end = timeutil.EndOfMonth(timeutil.StartOfMonth(end).AddDate(0, -1, 0))
 	}
-	end = timeutil.EndOfMonth(end)
```
Comment (swayne275) [inline vault/activity_log.go:1742]: do we still need to pin `end` to be the end of the provided `end` month in the case where `end` is not the current month?

e.g. what if `end` is in the middle of last month?

Comment (hellobontempo) [reply vault/activity_log.go:1742]: is "provided" end month here referencing the `end_date` query param? If so, in the UI we send the 20th of the month, as we expect the backend to re-adjust it to the last day of that month.

Comment (hellobontempo) [reply vault/activity_log.go:1742]: following up, appears to be something different - as the response looks correct on my end. disregard!

Comment (HridoyRoy) [reply vault/activity_log.go:1742]: Fixed!

**Review by swayne275 — APPROVED** [review]

**Pull request merged**